.github/workflows/test.yml

```diff
@@ -20,48 +20,37 @@ jobs:
         # environments.
         config:
         - os: windows-2019
-          vcvars: "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"
+          preamble: call "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvarsall.bat" x64
+          shell: cmd
         - os: ubuntu-20.04
+          shell: bash
         # Temporarily disabling macOS CI due to pybind issue, see:
         #   https://github.com/conan-io/conan-center-index/pull/11075
         # - os: macos-10.15
+    defaults:
+      run:
+          # Annoyingly required here since `matrix` isn't available in
+          # the `shell` property of individual steps.
+          shell: ${{ matrix.config.shell }}
 
     steps:
     - uses: actions/checkout@v2
 
     - name: Bootstrap
       uses: ./.github/bootstrap_platform
 
-    - name: Configure CMake build (Unix)
-      if: runner.os != 'Windows'
+    - name: Configure CMake build
       run: >
+        ${{ matrix.config.preamble }}
```

Comment:
[fix] Despite appearences, this [actually breaks CI](https://github.com/OpenAssetIO/OpenAssetIO/actions/runs/3313127660/jobs/5470685372). 

In YAML, a `>` means "join all lines into a single line, unless there is a blank line in-between".  So we should add a blank line between the preamble and `cmake` lines. Or alternatively use `|` instead of `>` and add `^`s to the multiline `cmake` invocation.

There is then of course the more concerning issue that the CI check still passes despite this causing an error :worried:

Reply:
:worried:  Oh wow, amazing spot. Possibly the whole workflow still has a bunch of bugs to work out then, :disappointed:

Reply:
Boy i've had a _fun time_ with some seemingly transient errors trying to get this fixed.

Addressed in [41e3197](https://github.com/OpenAssetIO/OpenAssetIO/pull/665/commits/41e319788184270f9ee2e1840a22a7133546d73e), later fixups address problems revealed from the windows tests actually executing.

Reply:
[plan] Perhaps worth an issue to look into why the CI step passed despite `vcvarsall` failing and if there's anything we can do about it.

Otherwise looks good and can resolve this.

Reply:
I'll just have a quick look into this the now rather than putting this off, if it becomes a rabbit hole will bail out and log something.

Reply:
Pretty difficult one to google. I had a look at the `vcvarsall` source itself.

Mostly what that script does is dispatch to other setup files, however, in the case that none of the dispatched scripts error, every return path from `vcvarsall` itself explicitly sets the error code to 0, even in the case where it reports a misuse error via `echo`

This is frustrating, we could scan the output for the word `error` I suppose, but that's seems a tad overly.

Reply:
OK no worries, I guess we just have to keep an eye out, thanks.